### PR TITLE
Post-purchase: Automatically cancel migration when user exits migration

### PR DESCRIPTION
**Pull request by gabrielcaires**
Related to #96115 
Depends on [#96103](https://github.com/Automattic/wp-calypso/pull/96103) 

## Proposed Changes
Cancel a migration (removing any pending stickers) in the following scenarios:
* Back button on `How To migrate` step
*  User selects `Import content only` on the `import-or-migrate` step. 

## Why are these changes being made?
*  We are facing multiple cases of the user's site having the pending label when they gave up on running a migration. 
The intent of this PR is to capture it and automatically cancel the migration, removing the pending label.

## Testing Instructions
* Enable the feature flag (DevTools> Console > and run:
`sessionStorage.setItem('flags,' 'automated-migration/pending-status')`
* Go to any migration flow 
* Follow all steps, choosing DIFM, until you reach the credential step
*  Use the back button to back to "How to Migrate"
* Open a new tab and go to `/sites` 
* Check your site is not marked as "Migration Pending"

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/data/site-migration/landing/use-migration-canceller/index.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	} );
+
+	return { status: 'success' };
+};
+
+interface Response {
+	status: 'success' | 'skipped';
+}
+
+export const useMigrationCanceller = ( siteId: SiteId | undefined ) => {
```
Comment (valterlorran) [inline client/data/site-migration/landing/use-migration-canceller/index.ts]: Nit: I think naming this `useMigrationCancellation` would be more clear.

**Review by valterlorran — COMMENTED** [review]
This is looking good.

I just think that the title of the PR could use some improvements to reflect that we are canceling the migration if the user exits the migration flow.

File: client/data/site-migration/landing/use-migration-canceller/index.ts (hunk 1 above)
Comment (andres-blanco) [reply client/data/site-migration/landing/use-migration-canceller/index.ts]: another name could be `useCancelMigration`

Comment (gabrielcaires) [reply client/data/site-migration/landing/use-migration-canceller/index.ts]: Renamed here [dc6c455](https://github.com/Automattic/wp-calypso/pull/96123/commits/dc6c45581d408138f158dd49fad1b49b71b52fb6)

Comment (gabrielcaires) [reply client/data/site-migration/landing/use-migration-canceller/index.ts]: Thinking a bit more,

We are not canceling a migration.
It seems more like abandoning a pending migration than canceling.

🤔  (If we have better ideas, I can update the name in future PR.

**Review by andres-blanco — APPROVED** [review]
LGTM!

**Pull request merged**